### PR TITLE
Upgrade Microsoft identity library NuGet packages

### PR DESCRIPTION
**Pull request by damieng**
Upgrade to the latest version of the IdentityModel libraries from Microsoft:

- *Microsoft.IdentityModel.Protocols.OpenIdConnect* 5.5.0
- *System.IdentityModel.Tokens.Jwt* 5.5.0

### CODE REVIEW

**Comment by damieng** [comment]
Fixes #293

File: src/Auth0.AuthenticationApi/Auth0.AuthenticationApi.csproj [hunk 1]
```diff
@@ -2,9 +2,10 @@
   <Import Project="..\..\build\common.props" />
   <PropertyGroup>
     <AssemblyTitle>Auth0.AuthenticationApi</AssemblyTitle>
-    <TargetFrameworks>netstandard2.0;net452;netstandard1.4;</TargetFrameworks>
     <AssemblyName>Auth0.AuthenticationApi</AssemblyName>
     <PackageId>Auth0.AuthenticationApi</PackageId>
+    <TargetFrameworks>netstandard2.0;net452;netstandard1.4;</TargetFrameworks>
+    <NetStandardImplicitPackageVersion>2.0.3</NetStandardImplicitPackageVersion>
```
Comment (damieng) [inline src/Auth0.AuthenticationApi/Auth0.AuthenticationApi.csproj]: This was unfortunately required due to some internal housekeeping in System.IdentityModel.Tokens.Jwt that broke things without it in a minor.

**Comment by williamdenton** [comment]
Please ensure the javascript library is also fixed else you will break people

**Comment by damieng** [comment]
I thought 5.5 reverted the behavior to the same as 5.3?  Is it instead still subtly different?

**Comment by williamdenton** [comment]
for users happily running on the current version referencing 5.4, upgrading to 5.5 is a breaking change. Its breaking because Auth0's javascript can't validate the jwt if the kid changes.


you need to fix this issue in your javascript library https://github.com/auth0/node-jwks-rsa/issues/5

**Comment by williamdenton** [comment]
to resolve this for us we had to manually patch the js to this


```js
instance.getSigningKey = function (kid, cb) {
    instance.logger(`Fetching signing key for '${kid}'`);

     instance.getSigningKeys((err, keys) => {
      if (err) {
        return cb(err);
      }

      const key = keys.find((k) => k.x5t === kid || k.kid === kid);
      if (key) {
        return cb(null, key);
      }
      instance.logger(`Unable to find a signing key that matches '${kid}'`);
      return cb(new JwksClientOriginal.SigningKeyNotFoundError(`Unable to find a signing key that matches '${kid}'`));
    });
  };
```


the crucial line is
```js
const key = keys.find((k) => k.x5t === kid || k.kid === kid);
```

which is changed from
```js
const key = keys.find(k => k.kid === kid); 
```

**Comment by damieng** [comment]
I understand the issue you've linked to in isolation - I just don't see how that library would ever be attempting to work with a JWT that came from Auth0.Net - this library only checks the token it is given - it does not create ones that would be consumed elsewhere.

**Comment by williamdenton** [comment]
our problem stemmed from including this library on our STS server which is currently issuing tokens. We added this library to our service as we are in the process of migrating to auth0, but are still issuing jwts to our own clients.

The effect of adding it to our existing server meant that the `/well-known/jwks` was not usable by the javascript client.

Its a fairly niche edge case but its impact is quite large.

if you have other customers in a similar situation but who are stable on version 5.4, upgrading them to 5.5 will break all clients with currently valid jwts being validated by your javascript library.

**Comment by damieng** [comment]
@williamdenton I think that is quite an unusual edge case. I really want to get this update out as it includes some other fixes and could prevent people falling into the 5.4 trap as well.

I'll put a note in the changelog and try and figure out how we can move node-jwks-rsa forward - there's a whole lot of issues outstanding on that repo not least of all the most recent version seems to have introduced an unintentional breaking change.

**Pull request merged**